app/models/piece.rb

```diff
@@ -11,4 +11,42 @@ class Piece < ActiveRecord::Base
   scope :queens, -> {where(type: 'Queen') }
   scope :pawns, -> {where(type: 'Pawn') }
 
+  # define some helper methods for move validation
+  def raise_exceptions(position_2)
```

Comment:
I might rename this method to something like:

`ensure_reasonable_move!`.  The exclamation point indicates to the caller that it may raise an exception.